presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -744,6 +744,10 @@ protected Scope visitTable(Table table, Optional<Scope> scope)
                                 .collect(toImmutableList());
                     }
 
+                    if (fields.isEmpty()) {
+                        throw new SemanticException(MISSING_COLUMN, table, "Table '%s' does not have any columns", table.getName());
```

Comment:
This is actually a connector error and not a semantic error for the user's query.

I think we should add a new `INVALID_TABLE` user error (analogous to `INVALID_VIEW`). That check should go in `MetadataManager.getTableMetadata()`, so it covers all usages. We turn the check here into a `verify()` (if it happens, there is a bug elsewhere).